android/src/firebase/core/TitaniumFirebaseCoreModule.java

```diff
@@ -36,25 +41,78 @@ public TitaniumFirebaseCoreModule()
 	public void configure(@Kroll.argument(optional=true)KrollDict param)
 	{
 		if (param != null) {
+			String apiKey = "";
+			String databaseURL = "";
+			String projectID = "";
+			String storageBucket = "";
+			String applicationID = "";
+			String GCMSenderID = "";
 			FirebaseOptions.Builder options = new FirebaseOptions.Builder();
-			if (param.containsKey("APIKey")) {
-				options.setApiKey(param.getString("APIKey"));
-			}
-			if (param.containsKey("databaseURL")) {
-				options.setDatabaseUrl(param.getString("databaseURL"));
-			}
-			if (param.containsKey("projectID")) {
-				options.setProjectId(param.getString("projectID"));
-			}
-			if (param.containsKey("storageBucket")) {
-				options.setStorageBucket(param.getString("storageBucket"));
-			}
-			if (param.containsKey("applicationID")) {
-				options.setApplicationId(param.getString("applicationID"));
-			}
-			if (param.containsKey("GCMSenderID")) {
-				options.setGcmSenderId(param.getString("GCMSenderID"));
+
+			if (param.containsKey("file")) {
+				// open file and parse it
+				try {
+					JSONObject json = new JSONObject(loadJSONFromAsset(param.getString("file")));
+					JSONObject projectInfo = json.getJSONObject("project_info");
+					String packageName = TiApplication.getAppCurrentActivity().getPackageName();
+
+					if (projectInfo.has("storage_bucket")) {
+						storageBucket = projectInfo.getString("storage_bucket");
+					}
+					if (projectInfo.has("firebase_url")) {
+						databaseURL = projectInfo.getString("firebase_url");
+					}
+					if (projectInfo.has("project_number")) {
+						GCMSenderID = projectInfo.getString("project_number");
+					}
+					if (projectInfo.has("project_id")) {
+						projectID = projectInfo.getString("project_id");
+					}
+					if (json.has("client")) {
+						JSONArray clients = json.getJSONArray("client");
+						for (int i = 0,len = clients.length(); i < len; i++) {
```

Comment:
missing space after the comma. Can you lint the source? We use clang-format for all native projects, e.g.
```bash
clang-format -style=file -i path/to/*.java
```

Reply:
sure, made the changes, it changed quiet a lot that's why I didn't do it in the beginning